### DISCUSSION TITLE
Misalignment of child nodes in initial loading of Mindmap

### DISCUSSION THREAD

**Discussion by Avinash489** [SSShooter/mind-elixir-core#189] [Q&A]
I have integrated in my angular application, i am generating the JSON with my data in it with required format and passing to mind map as input but, while initial load child nodes are having misalignment issues with the root node, If I do any action from UI like add, update or delete node mindmap alignment issues are fixed and renders perfect mindmap. 

Anyone faced this kind of issues,
I would greatly appreciate any insights, suggestions, or solutions that community members can provide to help resolve this misalignment problem.

Thank you all for your time and support!

Am attaching screenshots of my issue 
![Screenshot 2023-06-21 153151](https://github.com/ssshooter/mind-elixir-core/assets/17921843/73da3fdf-ebbd-4d07-9b3a-61c14cb582ba)

If i do any action in UI , it re-generates properly 
Excepted result :
![Screenshot 2023-06-21 153415](https://github.com/ssshooter/mind-elixir-core/assets/17921843/07d2176b-7692-4d3d-90c8-bc9297a3248e)

**Comment by SSShooter** [comment]
Could you reproduce the problem on [https://codesandbox.io](https://codesandbox.io/s/mind-elixir-vue3-dtcq6u)?

**Reply by Avinash489** [reply]
Thanks for your response,  This issue dosen't show up in a new angular application, but i have integrated in my existing application where i had some css conflicts which i have fixed them all, but uncertainty this particular issue of misalignment is showing up only to the Child node which is having subchildren, so am wondering what may be the reason causing this kind of issue.
But, If i do any action from UI like add, update or delete, mindmap automatically regenerates without any misalignment issue. (Only happeing in intial load of mind map with data)

It would be really helpful, if you guide me through what may be the reasons causing this kind of issue

![new](https://github.com/ssshooter/mind-elixir-core/assets/17921843/dbd694a8-39db-45e7-80a3-f56df741f745)

**Reply by Avinash489** [reply]
Here is my entire page screenshot of Mindmap generation when it initial load with my data, where you can only children with sub-children have mis-alignment issues.

![image](https://github.com/ssshooter/mind-elixir-core/assets/17921843/a14656fa-cf08-4341-9440-ddec9c7a58d8)

**Reply by SSShooter** [reply]
maybe, it's caused by the unexcept div size?

**Reply by Avinash489** [reply]
no there is no restrictions for div size, its integrated in full page, no blocking for size.

**Reply by Avinash489** [reply]
Only the height and width of the nodes which has children having this issue, that too only in initial load of mindmap.

**Reply by SSShooter** [reply]
I don't have any idea about this bug for now, maybe you can see here to find what cause the problem
https://github.com/ssshooter/mind-elixir-core/blob/44a36f228f70ef2aac4b2c82f6d8c108043359b5/src/linkDiv.ts#L70

**Reply by Avinash489** [reply]
sure thanks, I'll go through this and let you know if it resolved.
